checkov/common/util/parser_utils.py

```diff
@@ -356,7 +357,7 @@ def get_current_module_index(full_path: str) -> Optional[int]:
     return tf_index
 
 
-def is_nested(full_path: TFDefinitionKeyType) -> bool:
+def is_nested(full_path: TFDefinitionKeyType | None) -> bool:
```

Comment:
why not optional ?

Reply:
it is the same `TFDefinitionKeyType | None` is the newer preferred syntax compared to using `Optional[TFDefinitionKeyType]`